VideoWeb/VideoWeb/Controllers/InternalEventController.cs

```diff
@@ -0,0 +1,113 @@
+using VideoWeb.EventHub.Models;
+using VideoWeb.Mappings;
+using VideoApi.Client;
+using VideoApi.Contract.Requests;
+using EventType = VideoWeb.EventHub.Enums.EventType;
+using Task = System.Threading.Tasks.Task;
+using VideoWeb.Contract.Responses;
+
+namespace VideoWeb.Controllers
+{
+    [Produces("application/json")]
+    [ApiController]
+    [Route("internalevent")]
+    [Authorize(AuthenticationSchemes = "InternalEvent")]
+    public class InternalEventController : ControllerBase
+    {
+        private readonly IVideoApiClient _videoApiClient;
+        private readonly IEventHandlerFactory _eventHandlerFactory;
+        private readonly IConferenceCache _conferenceCache;
+        private readonly ILogger<InternalEventController> _logger;
+        private readonly IMapperFactory _mapperFactory;
+
+        public InternalEventController(
+            IVideoApiClient videoApiClient,
+            IEventHandlerFactory eventHandlerFactory,
+            IConferenceCache conferenceCache,
+            ILogger<InternalEventController> logger,
+            IMapperFactory mapperFactory)
+        {
+            _videoApiClient = videoApiClient;
+            _eventHandlerFactory = eventHandlerFactory;
+            _conferenceCache = conferenceCache;
+            _logger = logger;
+            _mapperFactory = mapperFactory;
+        }
+
+        [HttpPost("ParticipantsAdded")]
+        [SwaggerOperation(OperationId = "ParticipantsAdded")]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType(typeof(string), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> ParticipantsAdded(Guid conferenceId, AddParticipantsToConferenceRequest request)
+        {
+            var participantMapper = _mapperFactory.Get<ParticipantRequest, Participant>();
+            var participantResponseMapper = _mapperFactory.Get<Participant, ParticipantResponse>();
+            List<Participant> participantsAdded = request.Participants.Select(participant => participantMapper.Map(participant)).ToList();
+
+            try
+            {
+                var conference = await _conferenceCache.GetOrAddConferenceAsync(conferenceId, () =>
+                {
+                    _logger.LogTrace("Retrieving conference details for conference: {ConferenceId}", conferenceId);
+                    return _videoApiClient.GetConferenceDetailsByIdAsync(conferenceId);
+                });
+                _logger.LogDebug("ParticipantsAdded called. ConferenceId: {ConferenceId}, ParticipantCount: {ParticipantCount}",
+                    conferenceId, request.Participants.Count);
+
+                
+                foreach (var participant in participantsAdded)
+                {
+                    conference.AddParticipant(participant);
+                }
+
+                await _conferenceCache.UpdateConferenceAsync(conference);
+
+                
+                foreach (var participant in participantsAdded)
```

Comment:
Is there any reason the cache needs to be updated before this. It would be nice to merge this look with the one at line 70

Reply:
This was more for 7558 and a specific implementation that I will probably try to avoid anyway. I will change this.